config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -416,6 +427,549 @@ type MetricsServerConfig struct {
+	// When omitted, no relabeling is performed and all metrics are sent as-is.
+	// Maximum of 10 relabeling rules can be specified.
+	// Each rule must have a unique name.
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	// +listType=map
+	// +listMapKey=name
+	WriteRelabelConfigs []RelabelConfig `json:"writeRelabelConfigs,omitempty"`
+}
+
+// RelabelConfig represents a relabeling rule.
+// +kubebuilder:validation:XValidation:rule="self.action in ['Replace', 'HashMod'] ? has(self.targetLabel) && self.targetLabel != '' : true",message="targetLabel is required when action is Replace or HashMod"
+type RelabelConfig struct {
+	// name is a unique identifier for this relabel configuration.
+	// Must contain only alphanumeric characters, hyphens, and underscores.
+	// Must be between 1 and 63 characters in length.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:XValidation:rule="self.matches('^[a-zA-Z0-9_-]+$')",message="must contain only alphanumeric characters, hyphens, and underscores"
+	Name string `json:"name,omitempty"`
+	// sourceLabels specifies which label names to extract from each series for this relabeling rule.
+	// Each entry must be a valid label name (non-empty).
+	// The values of these labels are joined together using the configured separator,
+	// and the resulting string is then matched against the regular expression for
+	// the replace, keep, or drop actions.
+	// If a referenced label does not exist on a series, Prometheus substitutes an empty string.
+	// When omitted, the rule operates without extracting source labels (useful for actions like labelmap).
+	// Maximum of 10 source labels can be specified, each between 1 and 128 characters.
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	// +kubebuilder:validation:items:MinLength=1
+	// +kubebuilder:validation:items:MaxLength=128
+	// +listType=set
+	SourceLabels []string `json:"sourceLabels,omitempty"`
+	// separator is the character sequence used to join source label values.
+	// Common examples: ";" (default), ",", "::", "|||".
+	// When omitted, defaults to ";" (semicolon).
+	// Must be between 1 and 5 characters in length when specified.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=5
+	Separator string `json:"separator,omitempty"`
+	// regex is the regular expression to match against the concatenated source label values.
+	// Must be a valid RE2 regular expression (https://github.com/google/re2/wiki/Syntax).
+	// When omitted, defaults to "(.*)" (matches everything).
+	// Must be between 1 and 1000 characters in length when specified.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=1000
+	Regex string `json:"regex,omitempty"`
+	// targetLabel is the target label name where the result is written.
+	// Required for Replace and HashMod actions (enforced by validation).
```

Comment:
We don't need to mention that this is enforced by validation
```suggestion
	// Required for Replace and HashMod actions.
```